main.ml

```diff
@@ -329,7 +344,21 @@ let choose_datapath ?(persistent = true) response =
   | (scheme, u) :: us -> return (Ok (scheme, u, "0"))
 
 (* Process a message *)
-let process root_dir name x =
+let process root_dir name =
+  let version = ref None in
+  let compat_in field rpc =
+    match !version, rpc with
+    | Some "3.0", R.Dict d ->
+        R.Dict (List.filter ~f:(fun (k,_) -> k <> field) d)
+    | _ -> rpc
+  in
+  let compat_out fields rpc =
+    match !version, rpc with
+    | Some "3.0", R.Dict d ->
```

Comment:
There are several references to "3.0". I think we should bind this and use this trick:

```
let pvs_version = "3.0"

match !version with
| Some v when v = pvs_version ->
| ..
```